db/db_impl.cc

```diff
@@ -3130,21 +3131,22 @@ Status DBImpl::IngestExternalFile(
       TEST_SYNC_POINT_CALLBACK("DBImpl::IngestExternalFile:NeedFlush",
                                &need_flush);
       if (status.ok() && need_flush) {
+        FlushOptions flush_opts;
+        flush_opts.allow_write_stall = true;
         if (immutable_db_options_.atomic_flush) {
-          mutex_.Unlock();
           autovector<ColumnFamilyData*> cfds;
           SelectColumnFamiliesForAtomicFlush(&cfds);
-          status = AtomicFlushMemTables(cfds, FlushOptions(),
+          mutex_.Unlock();
```

Comment:
Good catch. I see one place where mutex is not held when calling ```SelectColumnFamiliesForAtomicFlush``` - https://github.com/facebook/rocksdb/blob/8d7bc76f364a6a613faf57ccf58d2e961bef879d/db/db_impl_compaction_flush.cc#L591. Not related to this PR, but maybe something to look into later.

Reply:
Good catch. I need to fix it in another PR.